sharding/collation.go

```diff
@@ -104,13 +104,43 @@ func (c *Collation) ProposerAddress() *common.Address {
 
 // CalculateChunkRoot updates the collation header's chunk root based on the body.
 func (c *Collation) CalculateChunkRoot() {
-	// TODO: this needs to be based on blob serialization.
-	// For proof of custody we need to split chunks (body) into chunk + salt and
-	// take the merkle root of that.
-	chunkRoot := common.BytesToHash(c.body)
+	chunks := Chunks(c.body)             // wrapper allowing us to merklizing the chunks.
+	chunkRoot := types.DeriveSha(chunks) // merklize the serialized blobs.
 	c.header.data.ChunkRoot = &chunkRoot
 }
 
+// CalculatePOC calculates the Proof of Custody given the collation body and
+// some salt value.
```

Comment:
Elaborate on the word salt here perhaps with a link to the original mention of it.